usaspending_api/references/v2/views/autocomplete.py

```diff
@@ -136,23 +136,48 @@ def post(self, request):
 class RecipientAutocompleteViewSet(BaseAutocompleteViewSet):
 
     def post(self, request):
-        """Return all recipients matching the provided search text"""
+        """Return all Parents and Recipients matching the provided search text"""
 
         search_text, limit = self.get_request_payload(request)
 
-        queryset = LegalEntity.objects.all()
+        # Return Recipients with valid id entries
+        queryset = LegalEntity.objects.exclude(
+            recipient_unique_id__isnull=True).exclude(
+            recipient_unique_id__exact='')
 
-        queryset = queryset.annotate(similarity=Greatest(TrigramSimilarity('recipient_name', search_text),
-                                                         TrigramSimilarity('recipient_unique_id', search_text))).\
-            order_by('-similarity')
-
-        exact_match_queryset = queryset.filter(similarity=1.0)
-        if exact_match_queryset.count() > 0:
-            queryset = exact_match_queryset
+        # Filter based on search text
+        response = {}
 
-        column_names = ['legal_entity_id', 'recipient_name', 'recipient_unique_id']
+        # Search and filter for Parent Recipients
+        parents = queryset.annotate(
+            similarity=Greatest(
+                TrigramSimilarity('recipient_name', search_text),
+                TrigramSimilarity('parent_recipient_unique_id', search_text))
+        ).distinct().order_by('-similarity').filter(
+            parent_recipient_unique_id__in=F('recipient_unique_id'))[:limit]
+
+        # Search and filter for Recipients, excluding Parent Recipients
+        recipients = queryset.annotate(
+            similarity=Greatest(
+                TrigramSimilarity('recipient_name', search_text),
+                TrigramSimilarity('recipient_unique_id', search_text))
+        ).distinct().order_by('-similarity').exclude(
+            parent_recipient_unique_id__in=F('recipient_unique_id'))[:limit]
+
+        response['results'] = {
```

Comment:
unfortunately `icontains` doesn't help with sorting my relevancy. There is also a case with trigram where random text still generates results. That would also cause both lists to be populated, potentially. Has frontend confirmed they're able to work with this response to display them with the proper relevance order?